query/src/pipelines/processors/transforms/transform_block_compact.rs

```diff
@@ -101,12 +107,16 @@ impl Compactor for BlockCompactor {
         let block = blocks[size - 1].clone();
 
         // perfect block
-        if block.num_rows() >= self.min_row_per_block && block.num_rows() <= self.max_row_per_block
+        if (block.num_rows() >= self.min_row_per_block
+            && block.num_rows() <= self.max_row_per_block)
+            || block.memory_size() >= self.max_bytes_per_block
```

Comment:
seem a typo, `block.memory_size() <= self.max_bytes_per_block` instead?

Reply:
No, it means this block is large enough, so we should flush this out.

Reply:
oh, I see.